lib/l2meter/emitter.rb

```diff
@@ -197,10 +197,10 @@ def build_token(key, value)
     end
 
     def format_token(key, value)
-      case value
-      when TrueClass
+      case
+      when value == true && !configuration.output_true_false_nil_values?
```

Comment:
It seems we're always negating the `#.output_true_false_nil_values?` configuration when we use it. Maybe that's telling us we should flip the config around so it's "positive?" Maybe something like:

```ruby
config.compact_values?
```

Then we could default that to `true` for backwards compatibility. Thoughts?

Reply:
I'm not very bothered about the naming. I deliberately went with something very precise (and therefore verbose) because I couldn't think of a good name which summarised it. "Compact values" is quite good though - I'd probably defer to your naming given you're probably coming at this more as a user would, with fresh eyes (as opposed to me, who thought about it and implemented it).

And I see what you mean about always negating it. It does actually seem like this "compacting" of values is a bit unusual (for example scrolls [doesn't](https://github.com/asenchi/scrolls/blob/master/lib/scrolls/parser.rb#L11-L18) and [neither does](https://github.com/roidrage/lograge/blob/master/lib/lograge/formatters/key_value.rb#L16-L28) lograge, to pick two other libraries I've seen) so actually it makes sense to frame it as something _this_ library is opting _into_ rather than the other way round anyway.

So I will update it 👍

Reply:
Follow up question: would you reverse the tests so the non-compacting tests are in the top-level context and the compacting ones are nested? Given what I said about it being unusual it kind of follows...

Reply:
> would you reverse the tests so the non-compacting tests are in the top-level context and the compacting ones are nested?

Yeah, I think I'd opt for the default behavior in the top-level, with a context for each various configuration lever we have.

Reply:
See https://github.com/heroku/l2meter/pull/9/commits/5e373d5f2a7a38022ce9f1d8b2246fd604931a6f.